modules/data.atmosphere/R/met.process.R

```diff
@@ -109,13 +109,28 @@ met.process <- function(site, input_met, start_date, end_date, model,
   }
   
   # read in registration xml for met specific information
-  register.xml <- system.file(paste0("registration/register.", met, ".xml"), package = "PEcAn.data.atmosphere")
-  register     <- read.register(register.xml, con)
-  
+  if(input_met$source != "CUSTOM"){
+    register.xml <- system.file(paste0("registration/register.", met, ".xml"), package = "PEcAn.data.atmosphere")
+    register     <- read.register(register.xml, con)
+  }
   # first attempt at function that designates where to start met.process
   if (is.null(input_met$id)) {
-    stage <- list(download.raw = TRUE, met2cf = TRUE, standardize = TRUE, met2model = TRUE)
-    format.vars <- query.format.vars(bety = bety, format.id = register$format$id)  # query variable info from format id
+    
+    if(input_met$source == "CUSTOM"){
+      # internal file is provided
+      stage <- list(download.raw = FALSE, met2cf = FALSE, standardize = FALSE, met2model = TRUE)
+      format.vars <- list()
+      format.vars$lat <- NULL
+      format.vars$lat <- NULL
+      format.vars$lat <- NULL
+    } else {
+      if(is.null(model)){
+        stage <- list(download.raw = TRUE, met2cf = TRUE, standardize = TRUE, met2model = FALSE)
```

Comment:
you changed the code from original, is this correct to set met2model to TRUE/FALSE based model.

Reply:
is.null(model) == true mean we don't need to convert to model format, we just need the internal format,

is.null(model) == false mean to do the convert old way, from XML to model.